bond/bond_test.go

```diff
@@ -38,110 +39,99 @@ const (
 			"cniVersion": "%s",
 			"mode": "%s",
 			"failOverMac": 1,
-			"linksInContainer": true,
+			"linksInContainer": %s,
 			"miimon": "100",
-			"mtu": 1400,
+			"mtu": %s,
 			"links": [
 				{"name": "net1"},
 				{"name": "net2"}
 			]
 		}`
 	ActiveBackupMode = "active-backup"
 	BalanceTlbMode   = "balance-tlb"
+	DefaultMTU       = "1400"
 )
 
 var Slaves = []string{Slave1, Slave2}
 
 var _ = Describe("bond plugin", func() {
 	var podNS ns.NetNS
+	var linksInitNS ns.NetNS
+	var linkAttrs []netlink.LinkAttrs
 
-	BeforeEach(func() {
+	JustBeforeEach(func() {
```

Comment:
I understand what you're doing here with justbeforeeach and justbeforeafter, but I believe that this makes the code less readable. I think that for tests it doesn't matter much to have some duplication in the code, especially when it makes the code more readable.

What you can do is to create different container nodes (describe, context) for the sections that have common beforeeach, instead of having a global one.

What do you think?

Reply:
The reason I chose to use the  justbeforeeach and justbeforeafter, is to match the ["Separating Creation and Configuration" pattern mentioned in the ginkgo framework docs](https://onsi.github.io/ginkgo/#separating-creation-and-configuration-justbeforeeach). I think this pattern is an exact match for this situation. In this case  the justbeforeeach/justbeforeafter are in charge of doing the creation of the environment  and the beforeeach/beforeafter clauses only contain configuration. I think its good to use the framework with the recommended pattern. 

Also i believe its good to separate configuration and creation .When reading test cases the relevant info is only the config for the test case - no need to each time go over all the details of the creation. in our case for each testcase it only matters what are the attributes of the links (and the initial name space of the links) - no need to add the creation code that will mix with the config code - the part that is important for understanding the  specific case.

I think duplication in test code makes the code also less readable - because each time the reader gets to a test case he has to go over all the code again (he cant just assume its the same in all tests) so it also takes effort and in my opinion makes the code less readable.

I reviewed the code again did some changes. including removing the  justaftereach, turned it in to an aftereach and some more changes. Now i think the separation between config and creation is better and the code is more clear. 
what do you think? 


In case you still think its not clear i would like to understand your point of view better. Why "justbeforeeach, makes the code less readable" ? I want to understand the reason, Is it because there are 2 different functions that include setup code? or what bothers you is the "just" because you don`t like that its runs last but its written in the beginning of the code?
Asking so we can find the best solution :)